sprimber-engine/src/main/java/com/griddynamics/qa/sprimber/engine/executor/ErrorMapper.java

```diff
@@ -40,24 +40,29 @@ OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
 public class ErrorMapper {
 
     private final List<String> skippedExceptionNames;
+    private final List<String> failedExceptionNames;
     private final List<Class<? extends Annotation>> pendingExceptionAnnotations;
 
     public ErrorMapper(List<String> skippedExceptionNames,
-                       List<Class<? extends Annotation>> pendingExceptionAnnotations) {
+                       List<String> failedExceptionNames, List<Class<? extends Annotation>> pendingExceptionAnnotations) {
         this.skippedExceptionNames = skippedExceptionNames;
+        this.failedExceptionNames = failedExceptionNames;
         this.pendingExceptionAnnotations = pendingExceptionAnnotations;
     }
 
     public ExecutionResult parseThrowable(Throwable throwable) {
 
         if (isPendingException(throwable)) {
-            return PENDING;
+            return new ExecutionResult(Status.PENDING, throwable);
         }
         if (Arrays.binarySearch(skippedExceptionNames.toArray(), throwable.getClass().getName()) >= 0) {
```

Comment:
Is there any insurance that skippedExceptionNames is sorted in this moment of time?